File: 2/2.py
memo = {0:0, 1:1}

def fib(n):
    if not n in memo:
        memo[n] = fib(n-1) + fib(n-2)
    return memo[n]
evens = []

n = 0
while fib(n) < 4000000:
    if fib(n) % 2 == 0:
        evens.append(fib(n))
    n = n + 1

print sum(evens)

File: 2/2.cpp
#include <iostream>
#include <cmath>

int fib(int n){
    double const PHI = 1.61803398875;
    double const ROOT_FIVE = pow(5, 0.5);
    int result = round(((pow(PHI,n)) - (pow((1 - PHI),n)))/ROOT_FIVE);
    return result;
}

int main(int argc, const char * argv[])
{
    int i = 3;
    long current = fib(i);
    long sum = 0;
    while(current < 4000000)
    {
        sum = sum + current;
        i = i + 3;
        current = fib(i);
    }
    std::cout << sum <<"\n";
    return 0;
}
